contrib/test_jenkins.sh

```diff
@@ -306,6 +306,26 @@ get_rdma_device_ip_addr() {
 	echo $ipaddr
 }
 
+get_non_rdma_ip_addr() {
+    if ! which ibdev2netdev >&/dev/null
+    then
+        return
+    fi
+
+    #get the interface of the ip address that is the default gateway (pure Ethernet IPv4 address).
+    eth_iface=$(ip route show|sed -n 's/default via \(\S*\) dev \(\S*\)/\2/p'| sed -e 's/\s\+$//')
+
+    #the pure Ethernet interface should not appear in the ibdev2netdev output. it should not be an IPoIB or
+    #RoCE interface.
+    if [ -z "$(ibdev2netdev | grep -w ${eth_iface})" ] ; then
```

Comment:
```
if  ! ibdev2netdev|grep -qw "${eth_iface}"
then
    echo "Failed to retrieve an IP of a non IPoIB/RoCE interface"
    exit 1
fi
```

Reply:
can it be like this?

```
    if [ -n "$(ibdev2netdev | grep -w ${eth_iface})" ] ; then
        echo "Failed to retrieve an IP of a non IPoIB/RoCE interface"
        exit 1
    fi

    echo $(get_ifaddr ${eth_iface})
```

Reply:
better check grep return code rather than its output

Reply:
ok but then it should be without !
`if $(ibdev2netdev|grep -qw "${eth_iface}")`

because `ibdev2netdev|grep -qw "${eth_iface}"` doesn't return anything in case of pure eth address, the return status is 1, and if ! is there it will exit.

Reply:
ok
so it should be
`if ibdev2netdev|grep -qw "${eth_iface}"`
need to remove `$( )` because we want the return code and not the output